src/app/modules/auth/auth.middlewares.ts

```diff
@@ -53,3 +57,32 @@ export const denyRpSpStudentEmails: ControllerHandler<
         .json({ message: 'User not found' }),
     )
 }
+
+/**
+ * Logs all admin actions which change database state (i.e. non-GET requests)
+ * @returns next
+ */
+export const logAdminAction: ControllerHandler<
+  unknown,
+  unknown,
+  unknown
+> = async (req, res, next) => {
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+  const body = req.body
+  const method = req.method
+
+  if (req.method.toLowerCase() !== 'get') {
+    logger.info({
+      message: 'Admin attempting to make changes',
+      meta: {
+        action: 'logAdminAction',
+        method,
+        ...createReqMeta(req),
+        sessionUserId,
+        body,
```

Comment:
a couple of additional things we should log:
1. the form ID, necessary bc the idea is to be able to see a trail of changes for each form. most of the time the formId is in `req.params`, but we can't assume that it is, so you might need to allow for it to be `undefined`. actually maybe you can just log `req.params` so we can query the form ID as `meta.params.formId`
2. the query params. although we do log `urlWithQueryParams` via `createReqMeta`, it might be easier to see and query it in object form.

Reply:
added in logging for `formId`, but for the other query params, they are only defined in the specific subroutes, so we would need to apply logging at the level of each subroute in order to capture them (which is what I wanted to avoid as it would be quite messy). I think logging by `formId` should be sufficient for investigative purposes - some untangling of the various requests based on URL will be needed but we don't expect this to be too onerous?

Reply:
can't we just log `req.query` and `req.params`?

Reply:
no need to type them, just log only